### PR TITLE
Sync Feature Branch With Main 

### PR DESCRIPTION
**Pull request by gjermundgaraba**
## Description

This PR pulls all the changes from main into the feature branch.
The only change I made was to update a call to GetMsgTransfer (commented in changes ->)

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against the correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#pull-request-targeting)).
- [ ] Linked to GitHub issue with discussion and accepted design, OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/main/docs/build/building-modules/11-structure.md) and [Go style guide](../docs/dev/go-style-guide.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/main/testing/README.md#ibc-testing-package).
- [ ] Updated relevant documentation (`docs/`).
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Provide a [conventional commit message](https://github.com/cosmos/ibc-go/blob/main/docs/dev/pull-requests.md#commit-messages) to follow the repository standards.
- [ ] Include a descriptive changelog entry when appropriate. This may be left to the discretion of the PR reviewers. (e.g. chores should be omitted from changelog)
- [ ] Re-reviewed `Files changed` in the GitHub PR explorer.
- [ ] Review `SonarCloud Report` in the comment section below once CI passes.

### CODE REVIEW

File: e2e/tests/transfer/upgradesv1_test.go [hunk 1]
```diff
@@ -0,0 +1,357 @@
...
+			channelA.PortID,
+			channelA.ChannelID,
+			transfertypes.V2,
+			transferCoins,
+			chainAWallet.FormattedAddress(),
+			chainBWallet.FormattedAddress(),
+			s.GetTimeoutHeight(ctx, chainB),
+			0,
+			"",
+			transfertypes.Forwarding{},
```
Comment (gjermundgaraba) [inline e2e/tests/transfer/upgradesv1_test.go:309]: Just had to add forwarding here. Besides that, no changes.

**Review by chatton — APPROVED** [review]
LGTM! happy to do a merge commit once the tests (excluding wasm) was looking good! Thanks for handling this

**Review by crodriguezvega — APPROVED** [review]
Thank you for taking care of this, @gjermundgaraba!

**Comment by gjermundgaraba** [comment]
@chatton There are two (non-wasm) tests still failing, but even after updating local values I cannot reproduce.

**Comment by chatton** [comment]
@gjermundgaraba let's still go ahead and merge I think, the test is passing locally for me also. It's annoying but I think we can get this merged to keep progress going on ics20-v2

**Pull request merged**